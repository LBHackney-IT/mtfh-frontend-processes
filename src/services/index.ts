export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
