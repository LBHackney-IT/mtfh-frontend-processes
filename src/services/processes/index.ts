export * from "./sole-to-joint";
export * from "./change-of-name";
