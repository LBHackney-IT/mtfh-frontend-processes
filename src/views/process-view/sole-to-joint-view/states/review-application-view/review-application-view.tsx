import React, { useState } from "react";

import { CloseProcessView } from "../../../../../components";
import { locale } from "../../../../../services";
import { IProcess } from "../../../../../types";
import { isCurrentState } from "../../../../../utils/processUtil";
import { DesBox } from "../../../shared/process-components";
import { SubmitCaseView } from "../../../shared/submit-case-view";
import { HoReviewFailedView } from "../ho-review-view/ho-review-failed-view";
import { HoReviewView } from "../ho-review-view/ho-review-view";
import { NewTenancyView } from "../new-tenancy-view/new-tenancy-view";
import {
  EligibilityChecksPassedBox,
  TenureInvestigationRecommendationBox,
} from "../shared";
import { TenureInvestigationView } from "../tenure-investigation-view";

import { Process } from "@mtfh/common/lib/api/process/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Box,
  Center,
  ErrorSummary,
  Spinner,
  StatusErrorSummary,
  StatusHeading,
} from "@mtfh/common/lib/components";

const { views } = locale;

interface ReviewApplicationViewProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  optional?: any;
}

export const ReviewApplicationView = ({
  processConfig,
  process,
  mutate,
  optional,
}: ReviewApplicationViewProps): JSX.Element => {
  const [globalError, setGlobalError] = useState<number>();
  const [documentsSigned, setDocumentsSigned] = useState<boolean>(false);
  const { submitted, closeCase, setCloseCase } = optional;
  const { data: tenure, error } = useTenure(process.targetId);
  const {
    applicationSubmitted,
    tenureInvestigationFailed,
    tenureInvestigationPassed,
    tenureInvestigationPassedWithInt,
    hoApprovalPassed,
    hoApprovalFailed,
    tenureAppointmentScheduled,
    tenureAppointmentRescheduled,
    interviewScheduled,
    interviewRescheduled,
    tenureUpdated,
    processClosed,
  } = processConfig.states;

  if (error) {
    return (
      <ErrorSummary
        id="tenure-investigation-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!tenure) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const tenant = tenure?.householdMembers.find((m) => m.isResponsible);

  if (applicationSubmitted.state === process.currentState.state && submitted) {
    return (
      <SubmitCaseView
        processConfig={processConfig}
        process={process}
        mutate={mutate}
        optional={optional}
      />
    );
  }

  return (
    <div data-testid="soletojoint-ReviewApplication">
      {globalError && (
        <StatusErrorSummary id="review-application-global-error" code={globalError} />
      )}
      <EligibilityChecksPassedBox />
      <DesBox
        title={views.submitCase.supportingDocumentsApproved}
        description={views.submitCase.viewDocumentsOnDes}
      />

      {!isCurrentState(applicationSubmitted.state, process) && (
        <TenureInvestigationRecommendationBox
          processConfig={processConfig}
          process={process}
        />
      )}

      {applicationSubmitted.state === process.currentState.state && (
        <TenureInvestigationView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {[
        tenureInvestigationFailed.state,
        tenureInvestigationPassed.state,
        tenureInvestigationPassedWithInt.state,
        interviewScheduled.state,
        interviewRescheduled.state,
      ].includes(process.currentState.state) && (
        <HoReviewView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {(isCurrentState(hoApprovalFailed.state, process) ||
        isCurrentState(processClosed.state, process)) && (
        <HoReviewFailedView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
        />
      )}

      {[
        hoApprovalPassed.state,
        tenureAppointmentScheduled.state,
        tenureAppointmentRescheduled.state,
        tenureUpdated.state,
      ].includes(process.currentState.state) && (
        <NewTenancyView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{
            closeCase,
            setCloseCase,
            documentsSigned,
            setDocumentsSigned,
            tenant,
          }}
        />
      )}

      {closeCase && (
        <>
          <Box variant="warning">
            <StatusHeading
              variant="warning"
              title={views.closeProcess.soleToJointClosed}
            />
          </Box>
          <CloseProcessView
            process={process}
            processConfig={processConfig}
            mutate={mutate}
            setGlobalError={setGlobalError}
          />
        </>
      )}
    </div>
  );
};
