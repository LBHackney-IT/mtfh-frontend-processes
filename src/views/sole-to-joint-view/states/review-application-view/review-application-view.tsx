import { useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { SoleToJointHeader } from "../../../../components";
import { locale } from "../../../../services";
import { IProcess } from "../../../../types";
import { CloseProcessView } from "../../shared/close-process-view";
import { HoReviewFailedView } from "../ho-review-view/ho-review-failed-view";
import { HoReviewView } from "../ho-review-view/ho-review-view";
import { NewTenancyView } from "../new-tenancy-view/new-tenancy-view";
import { DesBox, EligibilityChecksPassedBox, TenantContactDetails } from "../shared";
import { TenureInvestigationView } from "../tenure-investigation-view";

import { Process } from "@mtfh/common/lib/api/process/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Box,
  Center,
  ErrorSummary,
  Link,
  Spinner,
  StatusErrorSummary,
  StatusHeading,
  Text,
} from "@mtfh/common/lib/components";
import { BoxVariant } from "@mtfh/common/lib/components/box";
import { StatusHeadingVariant } from "@mtfh/common/lib/components/status-heading";

const { views } = locale;

interface ReviewApplicationViewProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  optional?: any;
}
type Recommendation = "approve" | "decline" | "int";
const isCurrentState = (state, process) => state === process.currentState.state;
const isPreviousState = (state, process) =>
  process.previousStates.find((previousState) => state === previousState.state);

const getRecommendation = (
  processConfig: IProcess,
  process: Process,
): {
  recommendation: Recommendation;
  recommendationBoxVariant: BoxVariant;
  recommendationHeadingVariant: StatusHeadingVariant;
} => {
  const { tenureInvestigationFailed, tenureInvestigationPassed } = processConfig.states;
  if (
    isCurrentState(tenureInvestigationFailed.state, process) ||
    isPreviousState(tenureInvestigationFailed.state, process)
  ) {
    return {
      recommendation: "decline",
      recommendationBoxVariant: "warning",
      recommendationHeadingVariant: "warning",
    };
  }
  if (
    isCurrentState(tenureInvestigationPassed.state, process) ||
    isPreviousState(tenureInvestigationPassed.state, process)
  ) {
    return {
      recommendation: "approve",
      recommendationBoxVariant: "success",
      recommendationHeadingVariant: "success",
    };
  }
  return {
    recommendation: "int",
    recommendationBoxVariant: undefined,
    recommendationHeadingVariant: "base",
  };
};

export const ReviewApplicationView = ({
  processConfig,
  process,
  mutate,
  optional,
}: ReviewApplicationViewProps): JSX.Element => {
  const [globalError, setGlobalError] = useState<number>();
  const { closeCase, setCloseCase } = optional;
  const { data: tenure, error } = useTenure(process.targetId);
  const {
    applicationSubmitted,
    tenureInvestigationFailed,
    tenureInvestigationPassed,
    tenureInvestigationPassedWithInt,
    hoApprovalPassed,
    hoApprovalFailed,
    tenureAppointmentScheduled,
    tenureAppointmentRescheduled,
    interviewScheduled,
    interviewRescheduled,
    tenureUpdated,
    processClosed,
  } = processConfig.states;
  const {
    recommendation,
    recommendationBoxVariant,
    recommendationHeadingVariant,
  }: {
    recommendation: Recommendation;
    recommendationBoxVariant: BoxVariant;
    recommendationHeadingVariant: StatusHeadingVariant;
  } = getRecommendation(processConfig, process);

  if (error) {
    return (
      <ErrorSummary
        id="tenure-investigation-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!tenure) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const tenant = tenure?.householdMembers.find((m) => m.isResponsible);

  return (
    <div data-testid="soletojoint-ReviewApplication">
      <SoleToJointHeader processConfig={processConfig} process={process} />
      {globalError && (
        <StatusErrorSummary id="review-application-global-error" code={globalError} />
      )}
      <EligibilityChecksPassedBox />
      <DesBox title={views.submitCase.supportingDocumentsApproved} />

      {!isCurrentState(applicationSubmitted.state, process) && (
        <Box variant={recommendationBoxVariant}>
          <StatusHeading
            variant={recommendationHeadingVariant}
            title={views.tenureInvestigation.tenureInvestigatorRecommendation(
              recommendation,
            )}
          />
        </Box>
      )}

      {!closeCase &&
        ![
          tenureInvestigationFailed.state,
          tenureInvestigationPassed.state,
          tenureInvestigationPassedWithInt.state,
          interviewScheduled.state,
          interviewRescheduled.state,
          processClosed.state,
        ].includes(process.currentState.state) &&
        (tenant ? (
          <TenantContactDetails tenant={tenant} />
        ) : (
          <Text>Tenant not found.</Text>
        ))}

      {applicationSubmitted.state === process.currentState.state && (
        <TenureInvestigationView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
        />
      )}

      {[
        tenureInvestigationFailed.state,
        tenureInvestigationPassed.state,
        tenureInvestigationPassedWithInt.state,
        interviewScheduled.state,
        interviewRescheduled.state,
      ].includes(process.currentState.state) && (
        <HoReviewView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {(isCurrentState(hoApprovalFailed.state, process) ||
        isCurrentState(processClosed.state, process)) && (
        <HoReviewFailedView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
        />
      )}

      {[
        hoApprovalPassed.state,
        tenureAppointmentScheduled.state,
        tenureAppointmentRescheduled.state,
      ].includes(process.currentState.state) && (
        <NewTenancyView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ closeCase, setCloseCase }}
        />
      )}

      {isCurrentState(tenureUpdated.state, process) && (
        <Box variant="success">
          <StatusHeading
            variant="success"
            title={views.tenureInvestigation.tenancySigned}
          />
          <div
            style={{ marginLeft: 60, marginTop: 17.5 }}
            className="govuk-link lbh-link lbh-link--no-visited-state"
          >
            <Link as={RouterLink} to="#" variant="link">
              {views.tenureInvestigation.viewNewTenure}
            </Link>
          </div>
        </Box>
      )}

      {(closeCase || isCurrentState(tenureUpdated.state, process)) && (
        <>
          <Box variant="warning">
            <StatusHeading variant="warning" title={views.closeCase.soleToJointClosed} />
          </Box>
          <CloseProcessView
            process={process}
            processConfig={processConfig}
            mutate={mutate}
          />
        </>
      )}
    </div>
  );
};
