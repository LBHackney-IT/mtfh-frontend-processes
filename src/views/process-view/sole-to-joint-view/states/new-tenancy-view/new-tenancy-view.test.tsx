import {
  getContactDetailsV2,
  getReferenceDataV1,
  mockContactDetailsV2,
  mockProcessV1,
  patchProcessV1,
  render,
  server,
} from "@hackney/mtfh-test-utils";
import { screen, waitForElementToBeRemoved } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { locale, processes } from "../../../../../services";
import { ReviewApplicationView } from "../review-application-view/review-application-view";

let submitted = false;
const setSubmitted = () => {};

const options = {
  url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
  path: "/processes/:processName/:processId",
};

const mockTenureAppointmentSchedule = (appointmentDateTime) => {
  return {
    ...mockProcessV1,
    currentState: {
      ...mockProcessV1.currentState,
      processData: {
        formData: {
          appointmentDateTime,
        },
        documents: [],
      },
      state: "TenureAppointmentScheduled",
    },
  };
};

describe("tenure-investigation-view", () => {
  beforeEach(() => {
    jest.resetModules();
    submitted = false;
  });

  test("it renders NewTenancy view correctly for TenureAppointmentScheduled state", async () => {
    server.use(getReferenceDataV1({}, 200));
    server.use(getContactDetailsV2(mockContactDetailsV2));
    const { container } = render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockTenureAppointmentSchedule("2099-10-12T08:59:00.000Z"),
          processName: "soletojoint",
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted }}
      />,
      options,
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    await expect(
      screen.findByText(locale.views.tenureInvestigation.documentsSigned, {
        exact: false,
      }),
    ).resolves.toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  test("it renders TenureInvestigation view correctly for TenureAppointmentScheduled state, date has passed, submit fails", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    server.use(patchProcessV1({}, 500));
    render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={mockTenureAppointmentSchedule("2010-10-12T08:59:00.000Z")}
        mutate={() => {}}
        optional={{ submitted, setSubmitted }}
      />,
      options,
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    const documentsSigned = screen.getByText(
      locale.views.tenureInvestigation.documentsSigned,
    );
    expect(documentsSigned).toBeEnabled();
    await userEvent.click(documentsSigned);
    await userEvent.click(screen.getByText(locale.views.closeProcess.outcomeLetterSent));
    await userEvent.click(screen.getByText(locale.confirm));
    await expect(
      screen.findByText("There was a problem with completing the action", {
        exact: false,
      }),
    ).resolves.toBeInTheDocument();
  });

  test("it renders TenureInvestigation view correctly for TenureAppointmentRescheduled state", async () => {
    server.use(getReferenceDataV1({}, 200));
    server.use(getContactDetailsV2(mockContactDetailsV2));
    const { container } = render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          processName: "soletojoint",
          currentState: {
            ...mockProcessV1.currentState,
            state: "TenureAppointmentRescheduled",
            processData: {
              formData: {
                appointmentDateTime: "2099-10-12T08:59:00.000Z",
              },
              documents: [],
            },
          },
          previousStates: [
            {
              ...mockProcessV1.currentState,
              state: "TenureAppointmentScheduled",
              processData: {
                formData: {
                  appointmentDateTime: "2010-10-12T08:59:00.000Z",
                },
                documents: [],
              },
            },
            {
              ...mockProcessV1.currentState,
              state: "HOApprovalPassed",
              processData: {
                formData: {
                  reason: "Test",
                },
                documents: [],
              },
            },
          ],
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted }}
      />,
      options,
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    await expect(
      screen.findByText(locale.views.tenureInvestigation.documentsSigned, {
        exact: false,
      }),
    ).resolves.toBeDisabled();
    expect(container).toMatchSnapshot();
  });
});
