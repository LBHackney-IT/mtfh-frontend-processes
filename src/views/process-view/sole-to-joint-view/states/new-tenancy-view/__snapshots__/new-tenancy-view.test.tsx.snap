// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tenure-investigation-view it renders NewTenancy view correctly for TenureAppointmentScheduled state 1`] = `
<div>
  <div
    data-testid="soletojoint-ReviewApplication"
  >
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Eligibility checks passed
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="padding-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          Show all eligibility checks
        </a>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Supporting documents approved
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="margin-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="http://des/teams/1/dashboard/requests"
          target="_blank"
        >
          View documents on the Document Evidence Store
        </a>
      </div>
    </div>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Tenure investigator recommendation: Interview Applicant
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Sole to joint tenure application approved
            </h4>
          </div>
        </div>
      </div>
    </div>
    <h2
      class="lbh-heading-h2"
    >
      Sole to joint tenure application approved, next steps:
    </h2>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment scheduled
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Monday 12th October 2099
            <br />
            Time: 
            08:59 am
          </p>
          <div
            style="display: flex; flex-direction: column; align-items: flex-start; margin-top: 15px;"
          >
            <button
              class="govuk-link lbh-link"
              type="button"
            >
              Change
            </button>
          </div>
        </div>
      </div>
    </div>
    <p
      class="lbh-body-s"
    >
      Global Web Consultant Alverta Denesik
       contact details:
    </p>
    <p
      class="lbh-body-s"
    >
      Phone:
      <span
        style="margin-left: 1em;"
      >
        075949281
      </span>
    </p>
    <p
      class="lbh-body-s"
    >
      Email:
      <span
        style="margin-left: 1em;"
      >
        Gay.Jacobi@hotmail.com
      </span>
    </p>
    <button
      class="govuk-button lbh-button"
      disabled=""
      type="button"
    >
      Documents signed
    </button>
  </div>
</div>
`;

exports[`tenure-investigation-view it renders TenureInvestigation view correctly for TenureAppointmentRescheduled state 1`] = `
<div>
  <div
    data-testid="soletojoint-ReviewApplication"
  >
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Eligibility checks passed
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="padding-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          Show all eligibility checks
        </a>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Supporting documents approved
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="margin-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="http://des/teams/1/dashboard/requests"
          target="_blank"
        >
          View documents on the Document Evidence Store
        </a>
      </div>
    </div>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Tenure investigator recommendation: Interview Applicant
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Sole to joint tenure application approved
            </h4>
          </div>
          <p
            class="lbh-body-m"
          >
            Test
          </p>
        </div>
      </div>
    </div>
    <h2
      class="lbh-heading-h2"
    >
      Sole to joint tenure application approved, next steps:
    </h2>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment missed
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Tuesday 12th October 2010
            <br />
            Time: 
            08:59 am
          </p>
        </div>
      </div>
    </div>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment scheduled
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Monday 12th October 2099
            <br />
            Time: 
            08:59 am
          </p>
          <div
            style="display: flex; flex-direction: column; align-items: flex-start; margin-top: 15px;"
          >
            <button
              class="govuk-link lbh-link"
              type="button"
            >
              Change
            </button>
          </div>
        </div>
      </div>
    </div>
    <p
      class="lbh-body-s"
    >
      Global Web Consultant Alverta Denesik
       contact details:
    </p>
    <p
      class="lbh-body-s"
    >
      Phone:
      <span
        style="margin-left: 1em;"
      >
        075949281
      </span>
    </p>
    <p
      class="lbh-body-s"
    >
      Email:
      <span
        style="margin-left: 1em;"
      >
        Gay.Jacobi@hotmail.com
      </span>
    </p>
    <button
      class="govuk-button lbh-button"
      disabled=""
      type="button"
    >
      Documents signed
    </button>
  </div>
</div>
`;
